src/mps/projmposum.jl

```diff
@@ -95,7 +71,7 @@ indices `(a,s1,s2,b)` to the space `(a',s1',s2',b')`
 then the size is `(d,d)` where
 `d = dim(a)*dim(s1)*dim(s1)*dim(b)`
 """
-Base.size(P::ProjMPOSum) = size(P.pm[1])
+size(A::AbstractSum) = size(first(terms(A)))
```

Comment:
Add a check `@assert allequal(size.(terms(P))`.

Reply:
(Note that `allequal` was introduced in Julia 1.8 but should be available with `Compat.jl`).